### PR TITLE
APM-313346 create sample app to generate metrics and logs

### PR DESCRIPTION
**Pull request by karolkieglerski**

### CODE REVIEW

File: tests/e2e/deployment-test.sh [hunk 1]
```diff
@@ -68,6 +68,16 @@ fi
 
 sudo mv linux-amd64/helm /usr/local/bin/helm
 
+# Create E2E Sample App
+gcloud functions deploy sample_app \
+--runtime python37 \
+--trigger-http \
+--source sample_app/ \
+--allow-unauthenticated
```
Comment (pgutkowski) [inline tests/e2e/deployment-test.sh:76]: With this Cloud Function allowing unauthenticated requests, we are telegraphing open endpoint for anyone to use. Some malicious actor could flood it with requests and make us a costly GCP bill.

Comment (pgutkowski) [reply tests/e2e/deployment-test.sh:76]: I'd suggest using this path https://cloud.google.com/functions/docs/securing/authenticating#authenticating_developer_testing to authenticate the requests.

File: tests/e2e/deployment-test.sh [hunk 2]
```diff
@@ -84,6 +83,26 @@ else
     gcloud pubsub subscriptions create "${PUBSUB_SUBSCRIPTION}" --topic="${PUBSUB_TOPIC}" --ack-deadline=120 --message-retention-duration=86400
 fi
 
+# Create Log Router Sink
+if [[ $(gcloud logging sinks  list --filter=name:"${LOG_ROUTER}" --format="value(name)") ]]; then
+    echo "Log Router [${LOG_ROUTER}] already exists, skipping"
+else
+  gcloud logging sinks create "${LOG_ROUTER}" "pubsub.googleapis.com/projects/${GCP_PROJECT_ID}/topics/${PUBSUB_TOPIC}" \
+    --log-filter='(resource.type="cloud_function" AND resource.labels.function_name="sample_app") OR resource.type="cloud_run_revision"' --description="Simple Sink for E2E tests"
```
Comment (sq2gxo) [inline tests/e2e/deployment-test.sh:91]: I think we don't want to include cloud_run_revision logs in this sink "OR resource.type="cloud_run_revision"?

**Review by pgutkowski — APPROVED** [review]

**Comment by karolkieglerski** [comment]
finished configuration permissions, ready for review

**Review by mswiatkowska — APPROVED** [review]

**Review by sq2gxo — APPROVED** [review]

**Pull request merged**